src/ai/backend/agent/vendor/linux.py

```diff
@@ -41,8 +56,13 @@ async def get_available_cores():
                     data = await resp.json()
                     return {idx for idx in range(data['NCPU'])}
         except aiohttp.ClientError:
+            num_htSiblings = await libnuma.htcpu_info()['num_htcpu']
             try:
-                return os.sched_getaffinity(os.getpid())
+                restricted_pid_cpu = os.sched_getaffinity(os.getpid())
+                for cpu in list(restricted_pid_cpu):
+                    if cpu > num_htSiblings - 1:
+                        restricted_pid_cpu.discard(cpu)
+                return restricted_pid_cpu
```

Comment:
I think we should just remove use of `os.sched_getaffinity(os.getpid())` entirely because this routine is for detecting the number of cores in containers.

Actually the activator and agent's `CPUPlugin` uses `libnuma.get_available_cores()`.
As `psutil.cpu_count()` provides `logical` option, I think we could just use it.

We also need to explicitly distinguish the number of cores reported for resource slots (including logical cores and multiplied by the overcommit factor) and license server (only physical cores) in the activator's `CPUPlugin`.